packages/codegen-ui/lib/utils/form-component-metadata.ts

```diff
@@ -31,14 +31,21 @@ export const mapFormMetadata = (form: StudioForm, formDefinition: FormDefinition
       }
       return fields;
     }, []),
-    onValidationFields: Object.entries(form.fields).reduce<{ [field: string]: FieldValidationConfiguration[] }>(
-      (validationFields, [field, config]) => {
-        const updatedFields = validationFields;
-        updatedFields[field] = ('validations' in config && config.validations) || [];
-        return updatedFields;
-      },
-      {},
-    ),
+    onValidationFields: Object.entries(formDefinition.elements).reduce<{
+      [field: string]: FieldValidationConfiguration[];
+    }>((validationFields, [elementName, elementConfig]) => {
+      const updatedValidationFields = validationFields;
```

Comment:
did you mean to spread into a new object here?

Reply:
No, this object is fine being mutated here. Just reassigning because of an ESLint rule (carrying over what was already there, actually)